lib/neo4j/active_node/has_n.rb

```diff
@@ -29,11 +29,16 @@ def inspect
 
       extend Forwardable
       %w(include? empty? count find first last ==).each do |delegated_method|
-        def_delegator :@enumerable, delegated_method
+        def_delegator :deferred_objects_or_enumerable, delegated_method
       end
 
       include Enumerable
 
+      def deferred_objects_or_enumerable
+        #This is not good enough. Updates have to be taken care of as well
+        @query_proxy.start_object.try(:new_record?) ? @deferred_objects : @enumerable
```

Comment:
Yes, the `try` is here because I don't understand under which conditions the start_object is actually present. This line is anyway very questionable as it does not take care of persisted dirty records which it should.